test/basic/src/test/java/io/narayana/lra/arquillian/resource/LRAMultipleParticipant1Initiator.java

```diff
@@ -0,0 +1,141 @@
+/*
+   Copyright The Narayana Authors
+   SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.narayana.lra.arquillian.resource;
+
+import static io.narayana.lra.arquillian.resource.LRAMultipleParticipant1Initiator.LRA_PARTICIPANT_PATH;
+import static org.eclipse.microprofile.lra.annotation.ParticipantStatus.Active;
+import static org.eclipse.microprofile.lra.annotation.ParticipantStatus.Compensated;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_ENDED_CONTEXT_HEADER;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.Type.REQUIRES_NEW;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.Response;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.eclipse.microprofile.lra.annotation.AfterLRA;
+import org.eclipse.microprofile.lra.annotation.Compensate;
+import org.eclipse.microprofile.lra.annotation.Forget;
+import org.eclipse.microprofile.lra.annotation.LRAStatus;
+import org.eclipse.microprofile.lra.annotation.Status;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+import org.jboss.logging.Logger;
+
+@ApplicationScoped
+@Path(LRA_PARTICIPANT_PATH)
+public class LRAMultipleParticipant1Initiator {
+    public static final String LRA_PARTICIPANT_PATH = "participant1-initiator";
+    public static final String TRANSACTIONAL_START_PATH = "start-work";
+    public static final String LRA_END_STATUS = "lra-end-status";
+    public static final String BASE_URL_PARAM = "base-url";
+
+    private static final Logger log = Logger.getLogger(LRAMultipleParticipant1Initiator.class);
+
+    private static final AtomicInteger forgetCount = new AtomicInteger(0);
+    private static final AtomicInteger compensateCount = new AtomicInteger(0);
+    private static final Map<URI, LRAStatus> status = new ConcurrentHashMap<>();
+
+    @GET
+    @Path(TRANSACTIONAL_START_PATH)
+    @LRA(value = REQUIRES_NEW, timeLimit = 200)
+    public Response start(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId, @QueryParam(BASE_URL_PARAM) URL baseUrl)
+            throws URISyntaxException {
+        log.infov("\033[0;1mStarting LRA with ID {0}", lraId);
+        log.infov("\033[0;1mReceived base URL from client: {0}", baseUrl);
+
+        WebTarget target = ClientBuilder.newClient()
```

Comment:
Having a closer look at your test I think you are involving the second participant in a wrong way, in fact if you want to have a second participant you can use the joinLRA method so that it is enlisted as a compensator for that LRA: https://github.com/jbosstm/lra/blob/61e9ed35c7c66acad4af11092476000a210fcca9/client/src/main/java/io/narayana/lra/client/internal/NarayanaLRAClient.java#L376.
I'll create a new test for this issue to explain you better what I mean.

Reply:
Sorry @arjantijms, my bad. Your reproducer is straightforward and correct, I'll resolve this conversation